Server/src/controllers/DataSetController.ts

```diff
@@ -0,0 +1,245 @@
+import { Request, Response } from 'express';
+import { DataSetService } from '../services/DataSetService';
+import { IDataRequestModel } from "../models/interfaces/DataRequestModelInterface";
+
+export class DataSetController {
+    private processedRequest: IDataRequestModel;
+    private dataSetService: DataSetService
+
+    constructor() {
+    }
+
+    /**
+     * This controller will take a request, verify that it is valid, and if it valid will then process the request,
+     * send it to the getDataService to acquire an array of data sets that match the entered search terms, and
+     * then return a response with the array of data sets. If the request is invalid, or some other problem arrises, 
+     * it will throw an error.
+     * 
+     * @param request
+     * An object containing the request information and parameters: Request 
+     * @param response 
+     * An object containing a response: Response
+     */
+    async createRequestForData(request: Request, response: Response) {

```

Comment:
very ambigous method name, unless it does something generic, but it doesnt, maybe call it createRequestForDataQuerying or requestDataByQuery

Reply:
Will change. This was one of the first controllers we created. We did not completely change the old code in the pre-existing dataset controller

Reply:
ok i see